Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Contract.cs

```diff
@@ -43,12 +43,20 @@ public static Contract Prepare(Guid customerId, int customerAge, int customerHei
             StandardDuration);
     }
 
-    public void Sign(DateTimeOffset signedAt, DateTimeOffset today)
+    internal BindingContract Sign(DateTimeOffset signedAt, DateTimeOffset dateNow)
     {
+        if (IsSigned)
+        {
+            throw new ContractAlreadySignedException();
+        }
+
         BusinessRuleValidator.Validate(
             new ContractCanOnlyBeSignedWithin30DaysFromPreparation(PreparedAt, signedAt));
 
         SignedAt = signedAt;
-        ExpiringAt = today.Add(Duration);
```

Comment:
Do not drop ExpiringAt - contract that has to be signed can also expire. 

Then, bindingContract would be like: 

```
BindingContract.Accept(Id, CustomerId, Duration);
```

and inside binding contract you will set its own expire at date

Reply:
Yeah that have to be different calculation. Let's focus on expiration binding contract in next PR. For contract I will keep as is.